internal/provisioner/eks_provisioner.go

```diff
@@ -320,24 +320,31 @@ func (provisioner *EKSProvisioner) UpgradeCluster(cluster *model.Cluster) error
 			return errors.Wrap(err, "failed to migrate EKS NodeGroup")
 		}
 
+		workerName := eksMetadata.ChangeRequest.WorkerName
```

Comment:
Should we check here `eksMetadata.ChangeRequest.WorkerName == eskMetadata.WorkerName` or does the `eskMetadata.NodeInstanceGroups` always change if there's a different `AMI` or pod number per node?

Reply:
Now we only manage one NodeGroup. And yes, `eskMetadata.NodeInstanceGroups` always changes.